cadence/transactions/bridge/bridge_nft_to_evm.cdc

```diff
@@ -41,11 +42,14 @@ transaction(nftContractAddress: Address, nftContractName: String, id: UInt64, re
         let vault = signer.storage.borrow<auth(FungibleToken.Withdrawable) &FlowToken.Vault>(
                 from: /storage/flowTokenVault
             ) ?? panic("Could not access signer's FlowToken Vault")
-        self.tollFee <- vault.withdraw(amount: FlowEVMBridgeConfig.bridgeFee)
+        self.tollFee <- vault.withdraw(amount: FlowEVMBridgeConfig.bridgeFee) as! @FlowToken.Vault
+        // Borrow a reference to the signer's COA
+        self.coa = signer.storage.borrow<&EVM.BridgedAccount>(from: /storage/evm)
+            ?? panic("Could not borrow COA from provided gateway address")
     }
 
     execute {
         // Execute the bridge
-        FlowEVMBridge.bridgeNFTToEVM(token: <-self.nft, to: self.evmRecipient, tollFee: <-self.tollFee)
+        self.coa.depositNFT(nft: <-self.nft, to: self.evmRecipient, fee: <-self.tollFee)
```

Comment:
does this mean every coa will become a bridge itself?

Reply:
If is a redirect call and not any of the data from the COA itself is used, it might feel confusing, I think it make sense if we don't need evmRecipient arg and use the coa ?

Reply:
In recent conversation with @dete we decided the 'to: EVM.EVMAddress' param should be removed from the signature and provided from the calling COA's 'self.address()'. Haven't gotten a chance to make that change yet.

Also, bridging from EVM requires a reference to the calling COA which is passed as &self from within the calling COA.

There are some changes made in the EVM contract to support this path. Curious what you think about them in the context of the calling transaction and updates to the COA interface given the attributes used for bridge requests